src/cycax/cycad/engines/cycax_part_freecad.py

```diff
@@ -205,43 +214,75 @@ def hole(
-        active_doc.activeView().viewTop()
+
+        target_image_file = f"{self.filepath}-{view}.png"
         active_doc.activeView().fitAll()
         active_doc.activeView().saveImage(str(target_image_file), 2000, 1800, "White")
 
-    def render_to_dxf(self, active_doc: App.Document, target_path: Path):
-        """This method will be used for creating a pdxf of the object currently in view.
+    def change_view(self, active_doc: FreeCADGui.activeDocument, side: str, default: str = None):
+        """This will change the gui view to show the specified side.
         Args:
-            active_doc: The active FreeCAD Gui
-            target_path: The Path the png needs to be saved under.
+            active_doc: Freecad active doc.
+            side: The side the view is from.
+            default: The default side for that view.
         """
+
+        if side is None:
+            side = default
+
+        match side.upper().strip():
+            case "TOP":
+                active_doc.activeView().viewTop()
+            case "BACK":
+                active_doc.activeView().viewRear()
+            case "REAR":
+                active_doc.activeView().viewRear()
+            case "BOTTOM":
+                active_doc.activeView().viewBottom()
+            case "FRONT":
+                active_doc.activeView().viewFront()
+            case "LEFT":
+                active_doc.activeView().viewLeft()
+            case "RIGHT":
+                active_doc.activeView().viewRight()
+            case "ALL":
+                active_doc.activeView().viewAxometric()
+            case _:
+                msg = f"side: {side} is not one of TOP, BOTTOM, LEFT, RIGHT, FRONT BACK OR ALL."
+                raise ValueError(msg)
+        return side
+
+    def render_to_dxf(self, view: str | None = None):
+        """This method will be used for creating a dxf of the object currently in view.
+        Args:
+            view: The side from which to produce the output file.
+        """
+        active_doc = FreeCADGui.activeDocument()
+        view = self.change_view(active_doc=active_doc, side=view, default="TOP")
+        FreeCADGui.SendMsgToActiveView("ViewFit")
         __objs__ = []
         __objs__.append(active_doc.getObject("Shape"))
 
-        importDXF.export(__objs__, str(f"{target_path}-perspective.dxf"))
+        importDXF.export(__objs__, str(f"{self.filepath}-{view}.dxf"))
 
-    def render_to_stl(self, active_doc: App.Document, target_path: Path):
+    def render_to_stl(self, active_doc: App.Document):
         """This method will be used for creating a STL of an object currently in view.
         Args:
             active_doc: The active FreeCAD Gui
-            target_path: The Path the png needs to be saved under.
         """
         for obj in active_doc.Objects:
             if obj.ViewObject.Visibility:
-                filename = f"{target_path}.stl"
+                filename = f"{self.filepath}.stl"
```

Comment:
FYI `self.filepath.with_prefix(".stl")` is the proper way.

Reply:
This gave an error and after a quick glance at the docs I couldn't see a similar method. I think I should go read the docs properly though.